src/vcf/validator.cpp

```diff
@@ -132,6 +132,65 @@ namespace ebi
         }
     }
 
+    bool is_compressed_file(std::istream &input,
+                            const std::string &source)
+    {
+        std::string message = "Input file is a compressed file ";
+
+        if (source != ebi::vcf::STDIN) {
+            boost::filesystem::path source_name(source);
+            std::string file_extension = source_name.extension().string();
+
+            if (file_extension == ".zip") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(zip)";
+                return true;
+            }
+            else if (file_extension == ".tar") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar)";
+                return true;
+            }
+            else if (file_extension == ".gz") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar.gz)";
+                return true;
+            }
+            else if (file_extension == ".xz") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar.xz)";
+                return true;
+            }
+            else if (file_extension == ".rar") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(rar)";
+                return true;
+            }
+        }
+
+        unsigned char magic[10];
+        input.read((char*)magic, sizeof(magic));
+
+        const unsigned char zip[4] = { 80, 75, 3, 4 };
+        const unsigned char tar_gz[2] = { 31, 139 };
+        const unsigned char tar_xz[5] = { 253, 55, 122, 88, 90 };
+        const unsigned char tar_z[5] = { 31, 157 };
+
+        if (std::equal(zip, zip + sizeof(zip), magic)) {
+            BOOST_LOG_TRIVIAL(warning) << message << "(zip)";
+            return true;
+        }
```

Comment:
this should also be extracted to a function

Reply:
sure .
I will extract this into a seperate function